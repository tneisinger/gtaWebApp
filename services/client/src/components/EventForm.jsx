import React from 'react';
import PropTypes from 'prop-types';

const EventForm = (props) => {
  if (props.formType === 'jobForm') {
    return (
      <form onSubmit={(event) => props.handleFormSubmit(event)}>

        <div className="form-group">
          <input
            name="client"
            className="form-control input-lg"
            type="text"
            placeholder="Client Name"
            required
            value={props.formData.jobForm.client}
            onChange={props.handleFormChange}
          />
        </div>

        <div className="form-group">
          <input
            name="description"
            className="form-control input-lg"
            type="text"
            placeholder="Description"
            required
            value={props.formData.jobForm.description}
            onChange={props.handleFormChange}
          />
        </div>

        <div className="form-group">
          <input
            name="amount_paid"
            className="form-control input-lg"
            type="number"
            placeholder="Amount Paid"
            required
            value={props.formData.jobForm.amount_paid}
            onChange={props.handleFormChange}
          />
        </div>

        <div className="form-group">
          <label htmlFor="paid_to">Paid To</label>
          <select
            id="paid_to"
            name="paid_to"
            className="form-control input-lg"
            placeholder="Paid To"
            required
            value={props.formData.jobForm.paid_to}
            onChange={props.handleFormChange}
          >
            <option value="Gladtime Audio">Gladtime Audio</option>
            <option value="Meghan">Meghan</option>
            <option value="Tyler">Tyler</option>
            <option value="Tyler/Meghan Separately">
              Tyler/Meghan Separately
            </option>
          </select>
        </div>

        <div className="form-group">
          <label htmlFor="worked_by">Worked By</label>
          <select
            id="worked_by"
            name="worked_by"
            className="form-control input-lg"
            required
            value={props.formData.jobForm.worked_by}
            onChange={props.handleFormChange}
          >
            <option value="Meghan">Meghan</option>
            <option value="Tyler">Tyler</option>
            <option value="Tyler and Meghan">Tyler and Meghan</option>
          </select>
        </div>

        <div className="form-group">
          <label htmlFor="confirmation">Confirmation</label>
          <select
            id="confirmation"
            name="confirmation"
            className="form-control input-lg"
            required
            value={props.formData.jobForm.confirmation}
            onChange={props.handleFormChange}
          >
            <option value="Confirmed">Confirmed</option>
            <option value="Pencilled In">Pencilled In</option>
          </select>
        </div>

        <div className="form-group">
          <label htmlFor="has_paid">Has Paid</label>
          <input
            id="has_paid"
            name="has_paid"
            className="form-check-input checkbox-lg"
            type="checkbox"
            value={props.formData.jobForm.has_paid}
            onChange={props.handleFormChange}
          />
        </div>

        <div className="form-group">
          <label htmlFor="start_date">Start Date</label>
          <input
            id="start_date"
            name="start_date"
            className="form-control input-lg"
            type="date"
            required
            value={props.formData.jobForm.start_date}
            onChange={props.handleFormChange}
          />
        </div>

        <div className="form-group">
          <label htmlFor="end_date">End Date</label>
          <input
            id="end_date"
            name="end_date"
            className="form-control input-lg"
            type="date"
            required
            value={props.formData.jobForm.end_date}
            onChange={props.handleFormChange}
          />
        </div>

      </form>
    )
  } else if (props.formType === 'oneTimeExpenseForm') {
    return (
      <form onSubmit={(event) => props.handleFormSubmit(event)}>

        <div className="form-group">
          <input
            name="merchant"
            className="form-control input-lg"
            type="text"
            placeholder="Bought From"
            required
            value={props.formData.oneTimeExpenseForm.merchant}
            onChange={props.handleFormChange}
          />
        </div>

        <div className="form-group">
          <input
            name="description"
            className="form-control input-lg"
            type="text"
            placeholder="Description"
            required
            value={props.formData.oneTimeExpenseForm.description}
            onChange={props.handleFormChange}
          />
        </div>

        <div className="form-group">
          <input
            name="amount_spent"
            className="form-control input-lg"
            type="number"
            placeholder="Amount Spent"
            required
            value={props.formData.oneTimeExpenseForm.amount_spent}
            onChange={props.handleFormChange}
          />
        </div>

        <div className="form-group">
          <label htmlFor="date">Date</label>
          <input
            id="date"
            name="date"
            className="form-control input-lg"
            type="date"
            required
            value={props.formData.oneTimeExpenseForm.date}
            onChange={props.handleFormChange}
          />
        </div>

        <div className="form-group">
          <label htmlFor="paid_by">Paid By</label>
          <select
            id="paid_by"
            name="paid_by"
            className="form-control input-lg"
            required
            value={props.formData.oneTimeExpenseForm.paid_by}
            onChange={props.handleFormChange}
          >
            <option value="Gladtime Audio">Gladtime Audio</option>
            <option value="Meghan">Meghan</option>
            <option value="Tyler">Tyler</option>
            <option value="Tyler/Meghan Separately">
              Tyler/Meghan Separately
            </option>
          </select>
        </div>

        <div className="form-group">
          <label htmlFor="tax_deductible">Tax Deductible</label>
          <input
            id="tax_deductible"
            name="tax_deductible"
            className="form-check-input checkbox-lg"
            type="checkbox"
            value={props.formData.oneTimeExpenseForm.tax_deductible}
            onChange={props.handleFormChange}
          />
        </div>

        <div className="form-group">
          <label htmlFor="category">Category</label>
          <select
            id="category"
            name="category"
            className="form-control input-lg"
            required
            value={props.formData.oneTimeExpenseForm.category}
            onChange={props.handleFormChange}
          >
            <option value="Business Equipment">Business Equipment</option>
            <option value="Business Supplies">Business Supplies</option>
            <option value="Gasoline">Gasoline</option>
            <option value="Vehicle Maintenance">Vehicle Maintenance</option>
            <option value="Travel Expense">Travel Expense</option>
            <option value="Entertainment">Entertainment</option>
            <option value="Food">Food</option>
          </select>
        </div>

      </form>
    )
  }
};

EventForm.propTypes = {
  formType: PropTypes.oneOf(['jobForm', 'oneTimeExpenseForm']).isRequired,
  handleFormSubmit: PropTypes.func.isRequired,
  formData: PropTypes.object.isRequired,
  handleFormChange: PropTypes.func.isRequired
};

export default EventForm;
