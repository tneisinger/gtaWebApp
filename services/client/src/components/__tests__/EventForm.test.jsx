import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

import EventForm, { emptyJobFormData, emptyOneTimeExpenseFormData,
         formTypes, formData } from '../EventForm';


const dummyFunction = () => { console.log('dummy function ran') };

formTypes.forEach((formType) => {
  test(`${formType} form renders properly`, () => {
    const component = <EventForm
                        formType={formType}
                        formData={formData}
                        handleFormSubmit={dummyFunction}
                        handleFormChange={dummyFunction}
                      />;
    const wrapper = shallow(component);
    const form = wrapper.find('form');
    expect(form.length).toBe(1);
    const formGroup = wrapper.find('.form-group');
    expect(formGroup.length).toBe(Object.keys(formData[formType]).length);
    expect(formGroup.get(0).props.children.props.name)
      .toBe(Object.keys(formData[formType])[0]);
    expect(formGroup.get(0).props.children.props.value).toBe('');
  });
  test(`${formType} form renders a snapshot properly`, () => {
    const component = <EventForm
                        formType={formType}
                        formData={formData}
                        handleFormSubmit={dummyFunction}
                        handleFormChange={dummyFunction}
                      />;
    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
