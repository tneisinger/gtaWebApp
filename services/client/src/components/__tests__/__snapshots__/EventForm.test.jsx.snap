// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Job form renders a snapshot properly 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="form-group"
  >
    <input
      className="form-control input-lg"
      name="client"
      onChange={[Function]}
      placeholder="Client Name"
      required={true}
      type="text"
      value=""
    />
  </div>
  <div
    className="form-group"
  >
    <input
      className="form-control input-lg"
      name="description"
      onChange={[Function]}
      placeholder="Description"
      required={true}
      type="text"
      value=""
    />
  </div>
  <div
    className="form-group"
  >
    <input
      className="form-control input-lg"
      name="amount_paid"
      onChange={[Function]}
      placeholder="Amount Paid"
      required={true}
      type="number"
      value=""
    />
  </div>
  <div
    className="form-group"
  >
    <label
      htmlFor="paid_to"
    >
      Paid To
    </label>
    <select
      className="form-control input-lg"
      id="paid_to"
      name="paid_to"
      onChange={[Function]}
      placeholder="Paid To"
      required={true}
      value="Gladtime Audio"
    >
      <option
        value="Gladtime Audio"
      >
        Gladtime Audio
      </option>
      <option
        value="Meghan"
      >
        Meghan
      </option>
      <option
        value="Tyler"
      >
        Tyler
      </option>
      <option
        value="Tyler/Meghan Separately"
      >
        Tyler/Meghan Separately
      </option>
    </select>
  </div>
  <div
    className="form-group"
  >
    <label
      htmlFor="worked_by"
    >
      Worked By
    </label>
    <select
      className="form-control input-lg"
      id="worked_by"
      name="worked_by"
      onChange={[Function]}
      required={true}
      value="Meghan"
    >
      <option
        value="Meghan"
      >
        Meghan
      </option>
      <option
        value="Tyler"
      >
        Tyler
      </option>
      <option
        value="Tyler and Meghan"
      >
        Tyler and Meghan
      </option>
    </select>
  </div>
  <div
    className="form-group"
  >
    <label
      htmlFor="confirmation"
    >
      Confirmation
    </label>
    <select
      className="form-control input-lg"
      id="confirmation"
      name="confirmation"
      onChange={[Function]}
      required={true}
      value="Confirmed"
    >
      <option
        value="Confirmed"
      >
        Confirmed
      </option>
      <option
        value="Pencilled In"
      >
        Pencilled In
      </option>
    </select>
  </div>
  <div
    className="form-group"
  >
    <label
      htmlFor="has_paid"
    >
      Has Paid
    </label>
    <input
      className="form-check-input checkbox-lg"
      id="has_paid"
      name="has_paid"
      onChange={[Function]}
      type="checkbox"
      value={false}
    />
  </div>
  <div
    className="form-group"
  >
    <label
      htmlFor="start_date"
    >
      Start Date
    </label>
    <input
      className="form-control input-lg"
      id="start_date"
      name="start_date"
      onChange={[Function]}
      required={true}
      type="date"
      value=""
    />
  </div>
  <div
    className="form-group"
  >
    <label
      htmlFor="end_date"
    >
      End Date
    </label>
    <input
      className="form-control input-lg"
      id="end_date"
      name="end_date"
      onChange={[Function]}
      required={true}
      type="date"
      value=""
    />
  </div>
</form>
`;

exports[`OneTimeExpense form renders a snapshot properly 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="form-group"
  >
    <input
      className="form-control input-lg"
      name="merchant"
      onChange={[Function]}
      placeholder="Bought From"
      required={true}
      type="text"
      value=""
    />
  </div>
  <div
    className="form-group"
  >
    <input
      className="form-control input-lg"
      name="description"
      onChange={[Function]}
      placeholder="Description"
      required={true}
      type="text"
      value=""
    />
  </div>
  <div
    className="form-group"
  >
    <input
      className="form-control input-lg"
      name="amount_spent"
      onChange={[Function]}
      placeholder="Amount Spent"
      required={true}
      type="number"
      value=""
    />
  </div>
  <div
    className="form-group"
  >
    <label
      htmlFor="date"
    >
      Date
    </label>
    <input
      className="form-control input-lg"
      id="date"
      name="date"
      onChange={[Function]}
      required={true}
      type="date"
      value=""
    />
  </div>
  <div
    className="form-group"
  >
    <label
      htmlFor="paid_by"
    >
      Paid By
    </label>
    <select
      className="form-control input-lg"
      id="paid_by"
      name="paid_by"
      onChange={[Function]}
      required={true}
      value="Gladtime Audio"
    >
      <option
        value="Gladtime Audio"
      >
        Gladtime Audio
      </option>
      <option
        value="Meghan"
      >
        Meghan
      </option>
      <option
        value="Tyler"
      >
        Tyler
      </option>
      <option
        value="Tyler/Meghan Separately"
      >
        Tyler/Meghan Separately
      </option>
    </select>
  </div>
  <div
    className="form-group"
  >
    <label
      htmlFor="tax_deductible"
    >
      Tax Deductible
    </label>
    <input
      className="form-check-input checkbox-lg"
      id="tax_deductible"
      name="tax_deductible"
      onChange={[Function]}
      type="checkbox"
      value={false}
    />
  </div>
  <div
    className="form-group"
  >
    <label
      htmlFor="category"
    >
      Category
    </label>
    <select
      className="form-control input-lg"
      id="category"
      name="category"
      onChange={[Function]}
      required={true}
      value="Business Equipment"
    >
      <option
        value="Business Equipment"
      >
        Business Equipment
      </option>
      <option
        value="Business Supplies"
      >
        Business Supplies
      </option>
      <option
        value="Gasoline"
      >
        Gasoline
      </option>
      <option
        value="Vehicle Maintenance"
      >
        Vehicle Maintenance
      </option>
      <option
        value="Travel Expense"
      >
        Travel Expense
      </option>
      <option
        value="Entertainment"
      >
        Entertainment
      </option>
      <option
        value="Food"
      >
        Food
      </option>
    </select>
  </div>
</form>
`;
